Firestore/src/Query.php

```diff
@@ -611,7 +619,7 @@ private function buildPosition($key, $fieldValues, $before)
                         DocumentSnapshot::class
                     ));
                 }
-                if (FieldValue::isSentinelValue($value)) {
+                if ($value instanceof FieldValueInterface) {
```

Comment:
In `snapshotPosition` further down, you also need to treat array-contains as an equality filter.